### PR TITLE
ECC: ECDH tests

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: host/xtest/xtest_4000.c [hunk 1]
```diff
@@ -4235,3 +4247,129 @@ static void xtest_tee_test_4008(ADBG_Case_t *c)
...
+	TEE_OperationHandle op;
+	TEE_ObjectHandle key_handle;
+	TEE_ObjectHandle sv_handle;
+	TEE_Attribute params[4];
+	size_t param_count = 0;
+	uint8_t out[2048];
+	size_t out_size;
+	uint32_t size_bytes;
+	int i;
+	struct derive_key_ecdh_t *pt = &(derive_key_ecdh[0]);
```
Comment (jenswi-linaro) [inline host/xtest/xtest_4000.c]: It looks like this value is never used.

File: host/xtest/xtest_4000.c [hunk 2]
```diff
@@ -4235,3 +4247,129 @@ static void xtest_tee_test_4008(ADBG_Case_t *c)
...
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+					&ret_orig)))
+		return;
+
+	for (i = 0; i < ARRAY_SIZE(derive_key_ecdh); i++) {
+		pt = &(derive_key_ecdh[i]);
```
Comment (jenswi-linaro) [inline host/xtest/xtest_4000.c]: A complicated way of writing `pt = derive_key_ecdh + i`. I know it's a matter of taste, but at least the parenthesis could be removed.

**Comment by jenswi-linaro** [comment]
With or without my comments addressed
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

File: host/xtest/xtest_4000.c (hunk 1 above)
Comment (pascal-brand-st-dev) [inline host/xtest/xtest_4000.c]: Right. It was only if there was nothing in the array, but it never happens, and it confuses the code. I'll remove the initialization.

File: host/xtest/xtest_4000.c (hunk 2 above)
Comment (pascal-brand-st-dev) [inline host/xtest/xtest_4000.c]: I'll remove the parenthesis.

**Pull request merged**